File: src/main/java/com/yerti/core/entity/ModelProtection.java
package com.yerti.core.entity;

import org.bukkit.entity.ArmorStand;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.event.player.PlayerInteractAtEntityEvent;

/**
 * Protects models from being damaged
 */
public class ModelProtection implements Listener {

    @EventHandler
    public void onEntityDamage(EntityDamageEvent event) {
        if (event.getEntity() instanceof ArmorStand) {
            ArmorStand stand = (ArmorStand) event.getEntity();
            if (stand.getCustomName().equals("CustomModelPart")) {
                System.out.println("Cancelling");
                event.setCancelled(true);
            }
        }
    }

    @EventHandler
    public void onEntityInteract(PlayerInteractAtEntityEvent event) {
        if (event.getRightClicked() instanceof ArmorStand) {
            ArmorStand stand = (ArmorStand) event.getRightClicked();
            if (stand.getCustomName().equals("CustomModelPart")) {

                event.setCancelled(true);
            }
        }
    }


}

File: src/main/java/com/yerti/core/entity/ModelProtection.kt
package com.yerti.core.entity

import org.bukkit.entity.ArmorStand
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.event.player.PlayerInteractAtEntityEvent

class ModelProtection : Listener {
    @EventHandler
    fun onEntityDamage(event: EntityDamageEvent) {
        if (event.entity is ArmorStand) {
            val stand = event.entity as ArmorStand
            if (stand.customName == "CustomModelPart") {
                println("Cancelling")
                event.isCancelled = true
            }
        }
    }

    @EventHandler
    fun onEntityInteract(event: PlayerInteractAtEntityEvent) {
        if (event.rightClicked is ArmorStand) {
            val stand = event.rightClicked as ArmorStand
            if (stand.customName == "CustomModelPart") {
                event.isCancelled = true
            }
        }
    }
}